### DISCUSSION TITLE
Support for Candeo C201 no neutral dimmer module

### DISCUSSION THREAD

**Discussion by IZZE2000** [Koenkk/zigbee2mqtt#18634] [Devices]
Hi!
Any plans for supporting the Candeo C201 no neutral (jippie!) dimmer module?
https://candeo.io/store/zigbee-dimmer-module-c201/

I does show up in zigbee2mqtt when pairing as not supported.

I assume it is very similar to the  Zigbee Candeo C204 LED smart dimmer switch but of course do not have any button,
https://candeo.io/store/zigbee-dimmer-module-with-or-without-neutral-c204/

I did look into the "Support new devices" section but i think it is a bit over my head.

Cheers

**Comment by IZZE2000** [comment]
I can actually see that it should be supported since it shows up as "manufName":"Candeo","modelId":"HK_DIM_A"
Zigbee2mqtt can see it but it shows it as an unsupported anyway?

Any ideas? Thanks!

{"id":57,"type":"Router","ieeeAddr":"0x385b44fffe1b3bce","nwkAddr":20167,"manufId":4107,"manufName":"Candeo","modelId":"HK_DIM_A","epList":[1,242],"endpoints":{"1":{"profId":260,"epId":1,"devId":257,"inClusterList":[0,3,4,5,6,8,2821,4096],"outClusterList":[25],"clusters":{"genBasic":{"attributes":{"modelId":"HK_DIM_A","manufacturerName":"Candeo","powerSource":178,"zclVersion":8,"appVersion":0,"stackVersion":0,"hwVersion":0,"dateCode":"20220531","swBuildId":"1.00"}}},"binds":[],"configuredReportings":[],"meta":{}},"242":{"profId":41440,"epId":242,"devId":97,"inClusterList":[],"outClusterList":[33],"clusters":{},"binds":[],"configuredReportings":[],"meta":{}}},"appVersion":0,"stackVersion":0,"hwVersion":0,"dateCode":"20220531","swBuildId":"1.00","zclVersion":8,"interviewCompleted":true,"meta":{},"lastSeen":1692539347510,"defaultSendRequestWhen":"immediate"}

**Comment by Koenkk** [comment]
Could you check if the issue is fixed with the following external converter: https://gist.github.com/Koenkk/d0ee09ae9b5796fe15586b8fe7d9f824

- save this as file next to `configuration.yaml` as `ext_converter.js`
- add it to `configuration.yaml`:
```yaml
external_converters:
  - ext_converter.js
```
- start z2m, check if issue is fixed

**Reply by IZZE2000** [reply]
Hi Koenkk,
Yes, thanks, now it is working :-)

**Reply by Koenkk** [reply]
Great, integrated the changes, you can remove the ext converter after updating to the next release.

**Reply by IZZE2000** [reply]
Great, sounds like music to my ears :-)
